Blog/gst-build-cross-build/cross-building-with-gstreamer.md

```diff
@@ -0,0 +1,320 @@
+
+This step will download the GStreamer repositories including some dependencies such as glib etc into the `subprojects` folder. Basically
+it tries to download as many "mesonified" third party libraries as possible,  and `breaking news` the cmake ones, as a bridge
+has been implemented recently if necessary.
+
+```
+# git clone https://gitlab.freedesktop.org/gstreamer/gst-build
+# cd gst-build
+# meson build_dir
+```
+
+```
+...
+
+All GStreamer modules 1.17.0.1
+
+  Subprojects
+                        FFmpeg: YES
+                         dssim: YES
+                    gl-headers: YES
+                      graphene: YES
+                  gst-devtools: YES
+          gst-editing-services: YES
+                  gst-examples: YES
+    gst-integration-testsuites: YES
+                     gst-libav: YES
+                       gst-omx: YES
+               gst-plugins-bad: YES
+              gst-plugins-base: YES
+              gst-plugins-good: YES
+                gst-plugins-rs: NO
+              gst-plugins-ugly: YES
+                    gst-python: NO
+               gst-rtsp-server: YES
+                     gstreamer: YES
+               gstreamer-sharp: Feature 'sharp' disabled
+               gstreamer-vaapi: YES
+                         gtest: NO
+                   libmicrodns: YES
+                       libnice: YES
+                        libpsl: YES
+                       libsoup: NO
+                      openh264: YES
+                           orc: YES
+                     pygobject: NO
+                        sqlite: YES
+                      tinyalsa: NO
+                          x264: YES
+Option buildtype is: debug [default: debugoptimized]
+Found ninja-1.8.2 at /usr/bin/ninja
+
+```
+
+After this steps, a new folder `build_dir` should be ready to be used by `ninja` to build the binaries.
+
+##### Build gst-build
+
+This step will build all GStreamer base libraries in addition to the plugins from base/good/bad/ugly/libav if their
+dependencies have been met or built by `gst-build` (ie glib, openh264 etc.).
+
+```
+# ninja -C build_dir
+```
+
+
+#### Test gst-build
+
+This command will create a new environment where all tools and plugins built previously are available in the env super-setting the system one with the right environment variables.
+
+```
+# ninja -C build_dir uninstalled
+```
+
+A prefix to  your prompt should be available as
+
+
+```
+[gst-master] bash-prompt #
+```
+
+```
+[gst-master] bash-prompt # env | grep GST_
+```
+
+From this environment you are now ready to use the power of GStreamer, and even implement new features in it without the fear of using out of date version.
+
+From this shell, you are also able to compile without exiting the environment. This feature is very convenient to test a branch or fix a bug. Go to the `subprojects` folder and modify the code directly.
+
+```
+[gst-master] bash-prompt # gst-inspect-1.0
+```
+
+### Let's Talk about Cross-Compilation:
```

Comment:
Wondering if it wouldn't be best to split this post in 2 articles : one about gst-build and the previous build tools, and a specific one about cross-compilation. The post is already quite long and cross-compilation is a specific topic on its own.